R/zchunk_L1233.Elec_water.R

```diff
@@ -39,89 +38,301 @@ module_water_L1233.Elec_water_DISABLED <- function(command, ...) {
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
-    #
-    # NOTE: there are 'match' calls in this code. You probably want to use left_join_error_no_match
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code uses vecpaste
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code uses repeat_and_add_vector
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # ===================================================
 
-    # Produce outputs
-    # Temporary code below sends back empty data frames marked "don't test"
-    # Note that all precursor names (in `add_precursor`) must be in this chunk's inputs
-    # There's also a `same_precursors_as(x)` you can use
-    # If no precursors (very rare) don't call `add_precursor` at all
-    tibble() %>%
-      add_title("descriptive title of data") %>%
-      add_units("units") %>%
-      add_comments("comments describing how data generated") %>%
-      add_comments("can be multiple lines") %>%
+    # STEP 1. DOWNSCALE WATER COOLING SYSTEM SHARES FROM 14 REGIONS to 201 COUNTRIES
+
+    # RENAME INTERMEDIATE FUELS AND AGGREGATE BY UPDATED FUEL TYPE
+    L101.en_bal_EJ_ctry_Si_Fi_Yh_full %>%
+      filter(sector == "out_electricity generation") %>%
+      left_join(select(enduse_fuel_aggregation, fuel, electricity), by = "fuel") %>%
+      select(-fuel) %>%
+      rename(fuel = electricity) -> L1233.out_EJ_ctry_elec_Fi_Yh_full
+
+    L1233.out_EJ_ctry_elec_Fi_Yh_full %>%
+      filter(year %in% HISTORICAL_YEARS) %>%
+      group_by(iso, year, fuel) %>%
+      summarise(value = sum(value)) %>%
+      ungroup() %>%
+      filter(is.na(fuel) == F) %>%
+      mutate(sector = "electricity generation") -> L1233.out_EJ_ctry_elec_F_Yh
+
+    # INTERPOLATE A23.CoolingSystemShares_RG3 FOR HISTORICAL YEARS AND HISTORICAL + FUTURE YEARS
+    A23.CoolingSystemShares_RG3 %>%
+      gather(year, value, -region_GCAM3, -plant_type, -cooling_system, -water_type) %>%
+      mutate(year = as.integer(year)) -> A23.CoolingSystemShares_RG3_LF
+
+    INTERPOLATED_YEARS <- HISTORICAL_YEARS[HISTORICAL_YEARS %in%
+                                              unique(A23.CoolingSystemShares_RG3_LF$year) == F]
+    EXTRAPOLATED_YEARS <- c(HISTORICAL_YEARS, FUTURE_YEARS)[c(HISTORICAL_YEARS, FUTURE_YEARS) %in%
+                                                              unique(A23.CoolingSystemShares_RG3_LF$year) == F]
+
+    setNames(data.frame(matrix(ncol = length(INTERPOLATED_YEARS),
+                               nrow = nrow(A23.CoolingSystemShares_RG3))),
+             as.character(INTERPOLATED_YEARS)) %>%
```

Comment:
Probably not necessary actually. I'm setting up a blank dataframe with headers equal to years I want to insert into a tibble. I then gather it and bind to the existing tibble before using the approx_fun to interpolate them. Is there a smarter way of inserting blank years (i.e., with NA values and repeating for all groups) into a tibble before interpolating? (As I write this out I'm thinking repeat_add_columns might do the job...).